docs/source/install.rst

```diff
@@ -0,0 +1,316 @@
+   be installed at this stage. Some (e.g. HDF5, hwloc) can also be installed
+   from source by PETSc during the ``configure`` :ref:`step<install_petsc>` by
+   passing additional flags (e.g. ``--download-hdf5``, ``--download-hwloc``).
+
+
+.. _install_petsc:
+
+Installing PETSc
+----------------
+
+For Firedrake to work as expected, it is important that a specific version of PETSc_
+is installed with a specific set of external packages. To install PETSc you need to
+do the following steps:
+
+#. Clone the PETSc repository::
+
+   $ git clone https://github.com/firedrakeproject/petsc.git
+   $ cd petsc
+
+#. Run PETSc ``configure``, passing in the flags generated by ``firedrake-configure``::
+
+   $ ./configure $(python3 ../firedrake-configure --show-petsc-configure-options)
+
+#. Compile PETSc::
+
+   $ make
+
+If you are using one of the
+:ref:`officially supported distributions<supported_systems>` then these configure
+options will include paths to system packages so PETSc can correctly find and
+link against them. If you are not then you should pass the ``--no-package-manager``
+flag to obtain a set of configure options where ``firedrake-configure``
+pessimistically assumes that no external packages are available, and hence need
+to be downloaded and compiled from source::
+
+   $ ./configure $(python3 ../firedrake-configure --no-package-manager --show-petsc-configure-options)
+
+For the ``default`` ARCH, running ``firedrake-configure`` with
+``--no-package-manager`` will produce the flags:
+
+.. literalinclude:: petsc_configure_options.txt
+   :language: text
+
+
+Customising the PETSc installation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Since ``firedrake-configure`` only outputs a string of options it is straightforward
+to make changes to the options passed to PETSc ``configure``. You can either:
+
+* Append additional options when ``configure`` is invoked. For example, to compile
+  PETSc with 64 bit indices you should run::
+
+   $ ./configure $(python3 ../firedrake-configure --show-petsc-configure-options) --with-64-bit-indices
+
+* Write the output of ``firedrake-configure`` to a file than can be modified::
+
+   $ python3 ../firedrake-configure --show-petsc-configure-options > my_configure_options.txt
+   <edit my_configure_options.txt>
+   $ ./configure $(cat my_configure_options.txt)
+
+
+.. _install_firedrake:
+
+Installing Firedrake
+--------------------
+
+Now that the right system packages are installed and PETSc is built we can now
+install Firedrake. To do this perform the following steps:
+
+#. Create a `virtual environment <https://docs.python.org/3/tutorial/venv.html>`_::
+
+      $ python3 -m venv venv-firedrake
+      $ . venv-firedrake/bin/activate
+
+   This is optional but strongly recommended to avoid polluting your system Python
+   environment.
+
+#. Set any necessary environment variables. This can be achieved using
+   ``firedrake-configure``::
+
+     $ export $(python3 firedrake-configure --show-env)
```

Comment:
Can you add a note to say that this _must_ be run from the directory that the `petsc` directory is in? Otherwise the `PETSC_DIR` environment variable will be incorrect. Not that I think many people would not do this, but it could lead to confusing errors if someone does.

Reply:
Fair point. And it's actually assuming the parent directory. PETSC_DIR is simply CWD/petsc

Reply:
I will also remove the `cd` bits from the install. Then it's less likely that people will be in the wrong place. I will still mention this though.

Reply:
Actually I can't. `./configure` cannot be run from another directory

Reply:
I have added a note on this (that I will push in a while, don't want to restart the workflows)

Reply:
Sounds good. Just a comment to reinforce why we need each `cd` command. Otherwise I'm sure there will be some smart-arse **cough** not me **cough** that goes "oh I can just run that from where I am, no need to `cd` just to `cd` back."

Reply:
I think it's a bit overly prescriptive to say "you MUST change directory or `configure` will not work". If you try it it will immediately fail. I don't want to overload the user with technical details